src/sentry/profiles/task.py

```diff
@@ -150,44 +157,33 @@ def _normalize(profile: Profile, organization: Organization) -> None:
     )
 
 
-@metrics.wraps("process_profile.symbolicate")
-def _symbolicate(profile: Profile, project: Project) -> None:
-    symbolicator = Symbolicator(project=project, event_id=profile["profile_id"])
+def _prepare_frames_from_profile(profile: Profile) -> Tuple[List[Any], List[Any]]:
     modules = profile["debug_meta"]["images"]
-    stacktraces = [
-        {
-            "registers": {},
-            "frames": s["frames"],
-        }
-        for s in profile["sampled_profile"]["samples"]
-    ]
+    if "frames" in profile["profile"]:
+        stacktraces = [{"registers": {}, "frames": profile["profile"]["frames"]}]
+    else:
+        stacktraces = [
+            {
+                "registers": {},
+                "frames": s["frames"],
+            }
+            for s in profile["sampled_profile"]["samples"]
+        ]
```

Comment:
This is indeed for compatibility. I had it as a check for a given version (`1`) but this is something that will likely stay the same so I switched it to checking for the key `version` and at that point, why not just check for the presence of the `frames` key instead.

I can switch back to check for the specific version (`1`) if you feel strongly about it otherwise I think it's fine the way it is. Perhaps I could add a comment too.